docs/features/kubernetes/permission-integration.md

```diff
@@ -0,0 +1,262 @@
+   | 'services'
+   | 'configmaps'
+   | 'deployments'
+   | 'limitranges'
+   | 'replicasets'
+   | 'horizontalpodautoscalers'
+   | 'jobs'
+   | 'cronjobs'
+   | 'ingresses'
+   | 'customresources'
+```
+
+- Ensure that the kubernetes resources in your cluster have a label that an entity in your catalog can use via label selector annotation. Workloads and custom endpoints make use of an entity reference field in the request, that entity reference should have an annotation like:
+
+```
+metadata:
+    annotations:
+        'backstage.io/kubernetes-label-selector': 'app=backstage'
+```
+
+that should correspond to the label you provide your kubernetes resource when creating it. In this case your pod should have the label 'app=backstage'
+
+- Ensure that you provide a backstage bearer token when calling the /workloads and /custom endpoints. The /clusters endpoint currently doesn't require this. Because a token is required for 2 out of the 3 endpoints it is required for you to login using one of the backstage auth providers. You can add a console.log line in packages/backend/src/plugins/auth.ts for an issued token under the provider that you are using to access that token.
+
+## How do the endpoints work prior to the permission framework integration?
+
+By default, Backstage endpoints are not protected, and all actions are available to anyone. The same is true for the Kubernetes plugin endpoints provided you have properly configured the cluster locator stanza in your [app-config](https://github.com/backstage/backstage/blob/master/app-config.yaml). By default if a user attempts to make a call to any of the following endpoints:
+
+- api/kubernetes/resources/workloads/query
+- api/kubernetes/resources/custom/query
+- api/kubernetes/clusters
+
+then you are returned a response like:
+
+### api/kubernetes/resources/workloads/query
+
+```
+{
+    items: [
+        {
+        cluster: {
+            name: 'clusterOne',
+        },
+        podMetrics: [],
+        resources: [
+            {
+            type: 'pods',
+            resources: [
+                {
+                metadata: {
+                    name: 'pod1',
+                },
+                },
+            ],
+            },
+            { type: 'services', resources: [] },
+            { type: 'configmaps', resources: [] },
+            { type: 'limitranges', resources: [] },
+            {
+            type: 'deployments',
+            resources: [
+                {
+                name: 'deployment1',
+                },
+            ],
+            },
+            { type: 'replicasets', resources: [] },
+            { type: 'horizontalpodautoscalers', resources: [] },
+            { type: 'jobs', resources: [] },
+            { type: 'cronjobs', resources: [] },
+            { type: 'ingresses', resources: [] },
+            { type: 'statefulsets', resources: [] },
+            { type: 'daemonsets', resources: [] },
+        ],
+        errors: [],
+        },
+    ],
+}
```

Comment:
Let's make this proper valid serialized JSON like you'd see in the output from curl or postman, with double quotes in all the right places:
```suggestion
{
  "items": [
    {
      "cluster": {
        "name": "clusterOne"
      },
      "podMetrics": [],
      "resources": [
        {
          "type": "pods",
          "resources": [
            {
              "metadata": {
                "name": "pod1"
              }
            }
          ]
        },
        { "type": "services", "resources": [] },
        { "type": "configmaps", "resources": [] },
        { "type": "limitranges", "resources": [] },
        {
          "type": "deployments",
          "resources": [
            {
              "name": "deployment1"
            }
          ]
        },
        { "type": "replicasets", "resources": [] },
        { "type": "horizontalpodautoscalers", "resources": [] },
        { "type": "jobs", "resources": [] },
        { "type": "cronjobs", "resources": [] },
        { "type": "ingresses", "resources": [] },
        { "type": "statefulsets", "resources": [] },
        { "type": "daemonsets", "resources": [] }
      ],
      "errors": []
    }
  ]
}
```